book/07-extending-learners.qmd

```diff
@@ -448,25 +541,24 @@ Some learners have a high number of parameters and it is easy to miss out on som
 In addition, if the maintainer of the upstream package changes something with respect to the arguments of the algorithm, the learner is in danger to break.
 Also, new arguments could be added upstream and manually checking for new additions all the time is tedious.
 
-Therefore we have written a "Parameter Check" that runs for every learner asynchronously to the R CMD Check of the package itself. This "Parameter Check" compares the parameters of the `r mlr3book::mlr_pkg("mlr3")` ParamSet against all arguments available in the upstream function that is called during `$train()` and `$predict()`. Again the file is automatically created and opened by `r ref("create_learner()")`, this will be named like `test_paramtest_package_type_key.R`, so in our example `test_paramtest_rpart_classif_rpart.R`.
+Therefore we have written a "Parameter Check" that runs regularly for every learner. This "Parameter Check" compares the parameters of the `r mlr3book::mlr_pkg("mlr3")` ParamSet against all arguments available in the upstream function that is called during `$train()` and `$predict()`. Again the file is automatically created and opened by `r ref("create_learner()")`, this will be named like `test_paramtest_package_type_key.R`, so in our example `test_paramtest_rpart_classif_rpart.R`.
+When the `.train` function calls multiple functions (e.g. a control function as described above), a list of functions can be passed to the parameter test.
 
 The test comes with an `exclude` argument that should be used to _exclude and explain_ why certain arguments of the upstream function are not within the ParamSet of the mlr3learner. This will likely be required for all learners as common arguments like `x`, `target` or `data` are handled by the `r mlr3book::mlr_pkg("mlr3")` interface and are therefore not included within the ParamSet.
```

Comment:
```suggestion
The test comes with an `exclude` argument that should be used to _exclude and explain_ why certain arguments of the upstream function are not within the ParamSet of the mlr3learner.
This will likely be required for all learners as common arguments like `x`, `target` or `data` are handled by the `r mlr3book::mlr_pkg("mlr3")` interface and are therefore not included within the ParamSet.
```